src/wp-includes/class-wp-scripts.php

```diff
@@ -333,7 +360,10 @@ public function do_item( $handle, $group = false ) {
 			 */
 			$srce = apply_filters( 'script_loader_src', $src, $handle );
 
-			if ( $this->in_default_dir( $srce ) && ( $before_handle || $after_handle || $translations_stop_concat ) ) {
+			// Used as a conditional to prevent script concatenation.
+			$is_deferred_or_async_handle = in_array( $strategy, array( 'defer', 'async' ), true );
+
+			if ( $this->in_default_dir( $srce ) && ( $before_handle || $after_handle || $translations_stop_concat || $is_deferred_or_async_handle ) ) {
```

Comment:
Readability suggestion

```suggestion
			if ( 
				$this->in_default_dir( $srce )
				&& ( $before_handle || $after_handle || $translations_stop_concat || $is_deferred_or_async_handle )
			) {
```

Reply:
Done.